src/lib/graphql_basic_scalars/graphql_basic_scalars.ml

```diff
@@ -0,0 +1,136 @@
+(**
+   This file defines basic graphql scalars in a shape usable by graphql_ppx for serialising.
+
+   It is meant to be used by backend graphql code.
+
+   The [grapqh_lib] library re-exports these basic scalars as well as other ones,
+   and is meant to be used by client code (via grapqh_ppx).
+ *)
+
+open Graphql_async.Schema
+
+module type S_JSON = sig
+  type t
+
+  val parse : Yojson.Basic.t -> t
+
+  val serialize : t -> Yojson.Basic.t
+
+  val typ : unit -> ('a, t option) Graphql_async.Schema.typ
+end
+
+let unsigned_scalar_scalar ~to_string typ_name =
+  scalar typ_name
+    ~doc:
+      (Core.sprintf
+         !"String representing a %s number in base 10"
+         (Stdlib.String.lowercase_ascii typ_name) )
+    ~coerce:(fun num -> `String (to_string num))
+
+module UInt32 : S_JSON with type t = Unsigned.UInt32.t = struct
+  type t = Unsigned.UInt32.t
+
+  let parse json = Yojson.Basic.Util.to_string json |> Unsigned.UInt32.of_string
+
+  let serialize value = `String (Unsigned.UInt32.to_string value)
+
+  let typ () =
+    unsigned_scalar_scalar ~to_string:Unsigned.UInt32.to_string "UInt32"
+end
+
+module UInt64 : S_JSON with type t = Unsigned.UInt64.t = struct
+  type t = Unsigned.UInt64.t
+
+  let parse json = Yojson.Basic.Util.to_string json |> Unsigned.UInt64.of_string
+
+  let serialize value = `String (Unsigned.UInt64.to_string value)
+
+  let typ () =
+    unsigned_scalar_scalar ~to_string:Unsigned.UInt64.to_string "UInt64"
+end
+
+module JSON = struct
+  type t = Yojson.Basic.t
+
+  let parse = Base.Fn.id
+
+  let serialize = Base.Fn.id
+
+  let typ () = scalar "JSON" ~doc:"Arbitrary JSON" ~coerce:serialize
+end
+
+module STRING : S_JSON with type t = string = struct
```

Comment:
Likewise `String` (or if you don't want to shadow the existing `String` module, `String_s` or `String_json`)